request_test.go

```diff
@@ -247,3 +248,113 @@ func TestOpendirHandleReuse(t *testing.T) {
 	rpkt = request.call(handlers, pkt, nil, 0)
 	assert.IsType(t, &sshFxpNamePacket{}, rpkt)
 }
+
+func Test_toLocalPath(t *testing.T) {
+	type args struct {
+		workDir string
+		p       string
+	}
+	tests := []struct {
+		name string
+		goos string
+		args args
+		want string
+	}{
+		{
+			name: "empty path with no workdir",
+			args: args{p: ""},
+			want: "",
+		},
+		{
+			name: "relative path with no workdir",
+			args: args{p: "file"},
+			want: "file",
+		},
+		{
+			name: "absolute path with no workdir",
+			args: args{p: "/file"},
+			want: "/file",
+		},
+		{
+			name: "workdir and empty path on Unix",
+			goos: "linux",
+			args: args{workDir: cleanPath("/home/user"), p: ""},
+			want: "/home/user",
+		},
+		{
+			name: "workdir and relative path on Unix",
+			goos: "linux",
+			args: args{workDir: cleanPath("/home/user"), p: "file"},
+			want: "/home/user/file",
+		},
+		{
+			name: "workdir and relative path with . on Unix",
+			goos: "linux",
+			args: args{workDir: cleanPath("/home/user"), p: "."},
+			want: "/home/user",
+		},
+		{
+			name: "workdir and relative path with . and file on Unix",
+			goos: "linux",
+			args: args{workDir: cleanPath("/home/user"), p: "./file"},
+			want: "/home/user/file",
+		},
+		{
+			name: "workdir and absolute path on Unix",
+			goos: "linux",
+			args: args{workDir: cleanPath("/home/user"), p: "/file"},
+			want: "/file",
+		},
+		{
+			name: "workdir and non-unixy path on Unix prefixes workdir",
+			goos: "linux",
+			args: args{workDir: cleanPath("/home/user"), p: "C:\\file"},
+			want: "/home/user/C:\\file",
+		},
+		{
+			name: "workdir and empty path on Windows",
+			goos: "windows",
+			args: args{workDir: cleanPath("C:\\Users\\User"), p: ""},
+			want: "C:\\Users\\User",
+		},
+		{
+			name: "workdir and relative path on Windows",
+			goos: "windows",
+			args: args{workDir: cleanPath("C:\\Users\\User"), p: "file"},
+			want: "C:\\Users\\User\\file",
+		},
+		{
+			name: "workdir and relative path with . on Windows",
+			goos: "windows",
+			args: args{workDir: cleanPath("C:\\Users\\User"), p: "."},
+			want: "C:\\Users\\User",
+		},
+		{
+			name: "workdir and relative path with . and file on Windows",
+			goos: "windows",
+			args: args{workDir: cleanPath("C:\\Users\\User"), p: "./file"},
+			want: "C:\\Users\\User\\file",
+		},
+		{
+			name: "workdir and absolute path on Windows",
+			goos: "windows",
+			args: args{workDir: cleanPath("C:\\Users\\User"), p: "/C:/file"},
+			want: "C:\\file",
+		},
+		{
+			name: "workdir and non-unixy path on Windows prefixes workdir",
+			goos: "windows",
+			args: args{workDir: cleanPath("C:\\Users\\User"), p: "C:\\file"},
+			want: "C:\\Users\\User\\C:\\file",
```

Comment:
I’d probably recommend adding a comment that while this looks like it’s a mistake, it is in fact correct behavior.

Since this is one of those fun things were the correct behavior looks like an obvious bug.